walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -51,34 +55,45 @@
         long duration = request.getDuration();
         int i = 0;
 
+        //while there is enough time for the requested meeting from the startTime of the timerange till the end of the day
         while(startTime + duration <= TimeRange.END_OF_DAY) {
-            int endTime = startTime;
 
+            int endTime = startTime;
             TimeRange currentEventTime = sortedEventsList.get(i).getWhen();
 
-            //currentEventTime.start() > startTime + duration checks if there is enough time for the meeting
-            while(currentEventTime.start() > startTime + duration && i < sortedEventsList.size()) {
+            while(i < sortedEventsList.size()) {
+                
+                currentEventTime = sortedEventsList.get(i).getWhen();
+
+                //if there is enough time to have the requested meeting from startTime to the beginning of the event we're currently looking at, we can create a new timerange
+                //from startTime - the end of the current event time
+                if(currentEventTime.start() >= startTime + duration) {
+                    endTime = currentEventTime.start();
+                    i++;
+                    break;
+                //if there is an overlap between the current event and the requested event, make the new startTime the end of the current event
+                } else if(currentEventTime.start() >= startTime && currentEventTime.start() < startTime + duration) {
+                    startTime = currentEventTime.end();
+                } 
+
                 i++;
-                //set the end time of the time range to the start of this meeting
-                endTime = currentEventTime.start();
-                if(i < sortedEventsList.size()) {
-                    currentEventTime = sortedEventsList.get(i).getWhen();
-                }
             }
 
             //if the endTime of the time range (which was set to startTime) is greater than or equal to the duration of the meeting, it can be added as a new time range
             if(endTime >= startTime + duration) {
                 availableTimes.add(TimeRange.fromStartEnd(startTime, endTime, false));
             }
 
-            //if we haven't looked at all events, set the next time range's start time to the end of the current event 
-            if(i < sortedEventsList.size()) {
-                startTime = currentEventTime.end();
-            }
+            //takes care of overlapping meetings
+            //for example, currentEvent = meeting 2
+            //meeting 1: 8:30am - 10am
+            //meeting 2: 9am - 9:30am
+            //we want the startTime of the next timerange to still be 10am, not 9:30am because meeting 1 is still going on
+            startTime = currentEventTime.end() > startTime ? currentEventTime.end() : startTime;
```

Comment:
Nice use of the ternary operator :)

Reply:
ternary operators are cool, but in this case, you can just use `Collections.max(Arrays.asList(currentEventTime.end(), startTime))`

Reply:
I changed it now so that it just uses `Math.max`.